web/app.js

```diff
@@ -687,11 +686,11 @@ var PDFViewerApplication = {
     var loadingTask = pdfjsLib.getDocument(parameters);
     this.pdfLoadingTask = loadingTask;
 
-    loadingTask.onPassword = function passwordNeeded(updatePassword, reason) {
-      PasswordPrompt.updatePassword = updatePassword;
-      PasswordPrompt.reason = reason;
-      PasswordPrompt.open();
-    };
+    loadingTask.onPassword = function passwordNeeded(callback, reason) {
+      this.passwordPrompt.setCallback(callback);
+      this.passwordPrompt.setReason(reason);
```

Comment:
Any reason, pun indented, to not just set both the callback *and* the reason using *one* function?